olive/model/__init__.py

```diff
@@ -488,7 +492,7 @@ def __init__(
         super().__init__(framework=Framework.PYTORCH, model_file_format=model_file_format, model_path=model_path)
 
         # io config for conversion to onnx
-        self.io_config = validate_config(io_config, IOConfig) if io_config else None
+        self.io_config = io_config if io_config and validate_config(io_config, IOConfig) else None
```

Comment:
if we want the dict, can you do `validate_config(io_config, IOConfig).dict() if io_config else None`? 

The io config validation does some conversions such as making `dynamic_axes` keys into int from string.

Reply:
Updated.